### PR TITLE
[ocilib] Add new port

### LINKED ISSUE #8187: [New Port Request] OCILIB
Library name: OCILIB

Library description: OCILIB is an open source and cross platform Oracle Driver that delivers efficient access to Oracle databases.

Source repository URL: https://github.com/vrogier/ocilib
Project homepage: https://vrogier.github.io/ocilib/

### PR DESCRIPTION
**Pull request by victorburckel**
Add new port for ocilib

- What does your PR fix? Fixes #8187

- Which triplets are supported/not supported? Have you updated the CI baseline?
x64-windows
x86-windows
x64-linux
CI baseline was updated to report 
x64-windows-static failure (no static build configuration in the sln)
arm64-windows, arm-uwp and x64-uwp failures

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/ocilib/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,47 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/ocilib/portfile.cmake:1]:
```suggestion
```
This function is deprecated. Could you please remove this line?

File: ports/ocilib/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+vcpkg_from_github(
+	OUT_SOURCE_PATH SOURCE_PATH
+	REPO vrogier/ocilib
+	REF v4.6.4
+	SHA512 83f5614a23c8fb4ab02517dec95d8b490c5ef472302735d5cc4cf483cc51513cc81ae2e1b4618c7c73fb5b071efe422e463b46fa79492ccb4775b511a943295a
+	HEAD_REF master
+	PATCHES
+		"out_of_source_build_version_file_configure.patch")
```
Comment (NancyLi1013) [inline ports/ocilib/portfile.cmake:10]:
```suggestion
       out_of_source_build_version_file_configure.patch
)
```

File: ports/ocilib/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		PLATFORM ${VCPKG_TARGET_ARCHITECTURE}
+		USE_VCPKG_INTEGRATION
+		ALLOW_ROOT_INCLUDES)
+
+	file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug)
+	file(COPY ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug)
+else()
+	vcpkg_configure_make(
+		SOURCE_PATH ${SOURCE_PATH}
+		OPTIONS --with-oracle-import=runtime)
```
Comment (NancyLi1013) [inline ports/ocilib/portfile.cmake:39]:
```suggestion
		OPTIONS 
		    --with-oracle-import=runtime
     )
```

File: ports/ocilib/CONTROL [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Source: ocilib
+Version: 4.6.4
+Homepage: https://vrogier.github.io/ocilib/
+Description: OCILIB is an open source and cross platform Oracle Driver that delivers efficient access to Oracle databases.
```
Comment (NancyLi1013) [inline ports/ocilib/CONTROL:4]:
```suggestion
Description: OCILIB is an open source and cross platform Oracle Driver that delivers efficient access to Oracle databases.
Supports: !(arm|uwp)
```
It seems that this port currently doesn't support `arm `and `uwp`. We need to add `Supports `field.

File: ports/ocilib/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [inline ports/ocilib/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")
````
Please also add this function.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by victorburckel** [comment]
Hi @NancyLi1013, thanks for the review. I did the suggested changes, but seems like there are some build failures with apr and apr-util on linux and osx.

**Comment by NancyLi1013** [comment]
The regressions for `apr `and `apr-util` have been fixed in #11559.
This is not related with this PR. You can ignore this. I'll rerun this PR once solved.

File: ports/ocilib/portfile.cmake (hunk 2 above)
Comment (NancyLi1013) [reply ports/ocilib/portfile.cmake]: Could you please address this review suggestion?

File: ports/ocilib/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/ocilib/portfile.cmake]: Could you please address this review suggestion?

**Comment by victorburckel** [comment]
Seems like the tests passed, let me now if you want me to do any other changes.

File: ports/ocilib/portfile.cmake (hunk 3 above)
Comment (victorburckel) [reply ports/ocilib/portfile.cmake]: Isn't what I did in the commit c928e7b (aka Code review) ok ?

File: ports/ocilib/portfile.cmake (hunk 2 above)
Comment (victorburckel) [reply ports/ocilib/portfile.cmake]: Isn't what I did in the commit c928e7b (aka Code review) ok ? I removed the quotes, and left the indent. Maybe you want me to switch from tabs to spaces for the indent ?

Comment (NancyLi1013) [reply ports/ocilib/portfile.cmake]: Yes, you're right. I have noticed you changed this. But I think the suggestion might be better.

File: ports/ocilib/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/ocilib/portfile.cmake]: In fact, both ways are fine with me. But we should put the right brackets to another line.

Comment (victorburckel) [reply ports/ocilib/portfile.cmake]: Ok my bad sorry, I didn't notice that the right parenthesis was on the next line. I aligned it with vcpkg_configure_make, I hope this is what you were thinking about.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/ocilib/portfile.cmake (hunk 2 above)
Comment (victorburckel) [reply ports/ocilib/portfile.cmake]: I put the parenthesis on the next line, I hope this is better.

**Comment by NancyLi1013** [comment]
@victorburckel 
Thanks for this PR.
LGTM now.

**Pull request merged**